.github/workflows/build.yaml

```diff
@@ -35,3 +38,58 @@ jobs:
           tags: "ghcr.io/${{ github.repository }}:${{ github.sha }}"
           cache-from: type=gha,scope=${{ github.repository }}
           cache-to: type=gha,scope=${{ github.repository }},mode=max
+
+  trivy-scan-code:
+    needs: build
+    name: Trivy Vulnerability Scan for Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Aqua Security Trivy Scan
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: 'fs'
+          security-checks: 'vuln,secret,config'
+          ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-results.sarif'
+          category: 'code'
+
+      - name: Image Scan ghcr
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'ghcr.io/${{ github.repository }}:${{ github.sha }}'
+          format: 'sarif'
+          output: 'trivy-image-results-ghcr.sarif'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+
+      - name: Image Scan redis
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'redis:alpine'
+          format: 'sarif'
+          output: 'trivy-image-results-redis.sarif'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+
+      - name: Upload Trivy Image Ghcr scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-image-results-ghcr.sarif'
+          category: 'image-ghcr'
+      
+      - name: Upload Trivy Image Redis scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-image-results-redis.sarif'
+          category: 'image-redis'
```

Comment:
Can you try translating this job into a reusable workflow? It will be useful for the tasks that will come next.

Here's the [doc](https://docs.github.com/en/actions/using-workflows/reusing-workflows#example-reusable-workflow)

You can create a new file called `.github/workflows/trivy.yaml` and call it with:
`uses: ./.github/workflows/trivy.yaml`

Reply:
I'm changing it